### PR TITLE
[test] Add PyTorch Horovod test

### PR DESCRIPTION
**Pull request by henrique**

### CODE REVIEW

**Comment by henrique** [comment]
rebased and all checks passing

**Comment by henrique** [comment]
@jenkins-cscs retry dom

**Comment by henrique** [comment]
@jenkins-cscs retry dom

**Review by rsarm — APPROVED** [review]

**Comment by henrique** [comment]
@jenkins-cscs retry daint

**Comment by henrique** [comment]
@jenkins-cscs retry daint

**Comment by henrique** [comment]
@jenkins-cscs retry daint

**Comment by henrique** [comment]
@jenkins-cscs retry daint

**Comment by henrique** [comment]
@jenkins-cscs retry daint

**Comment by henrique** [comment]
rebased again
can someone please merge this? 
thank you

**Comment by vkarak** [comment]
@henrique I had some style comments, but didn't have the time to make them yet. If you're in hurry, I'm fine merging this now.

File: cscs-checks/apps/pytorch/pytorch_horovod_check.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            'NCCL_IB_HCA': 'ipogif0',
+            'NCCL_IB_CUDA_SUPPORT': '1',
+            'OMP_NUM_THREADS': '$SLURM_CPUS_PER_TASK',
+        }
+        hash = 'master'
+        git_url = f'https://raw.githubusercontent.com/horovod/horovod/{hash}/examples/pytorch'  # noqa: E501
+        git_src = 'pytorch_synthetic_benchmark.py'
+        self.prerun_cmds = [
+            f'wget {git_url}/{git_src}',
+            f'echo starttime=`date +%s`'
```
Comment (vkarak) [inline cscs-checks/apps/pytorch/pytorch_horovod_check.py:39]: This this thing with the echos makes the time measurement a bit awkward. Couldn't you just `time` it? Also it seems that you are measuring also the creation of the virtual environment. Is this desired?

Comment (henrique) [reply cscs-checks/apps/pytorch/pytorch_horovod_check.py]: thanks for the input @vkarak 
that elapsed time wasn't actually being monitored, so I've removed it completely now
(the venv should also go away in the upcomming version of the pytorch recipe)

**Review by vkarak — APPROVED** [review]
lgtm

**Comment by vkarak** [comment]
@henrique There are random performance failures of the test during the CI. Can you do a series of runs and adjust the references to the median and they thresholds such that they cover the 95% CI ?

**Review by vkarak — APPROVED** [review]
lgtm

**Pull request merged**